### PR TITLE
Support multiple AWS accounts by setting AWS profiles

### PR DESCRIPTION
**Pull request by RemyDeWolf**
The user can now scan multiple AWS accounts by setting the AWS profiles:
```
# use the default region associated with each profile
cloudgrep --profiles dev-445935066876,prod-445935066876

# create 2 providers one for each profile, each with the regions specified
cloudgrep --profiles dev-445935066876,prod-445935066876 --regions us-east-1,us-west-2
```
The profile can also be set manually for each provider in the [config.yaml](https://github.com/run-x/cloudgrep/blob/remy/multi-account-aws-profiles/pkg/config/config.yaml#L41).
The default behavior doesn't change - it will use whatever profile is used by the current AWS credentials

Implementation notes:
- The AWS profile is set in AWS config [here](https://github.com/run-x/cloudgrep/blob/remy/multi-account-aws-profiles/pkg/provider/aws/provider.go#L79)
- The provider set the AccountId when converting the resources. [here](https://github.com/run-x/cloudgrep/blob/remy/multi-account-aws-profiles/pkg/resourceconverter/map_converter.go#L58) and [here](https://github.com/run-x/cloudgrep/blob/remy/multi-account-aws-profiles/pkg/resourceconverter/resource_converter.go#L71).
- The integration tests were updated to test for the AWS account id.
- Because we are adding a field in the data model - the demo DB was updated. The demo data was also updated to have 2 AWS distinct accounts. Here is the screenshot with the demo data and the new field account_idScreen Shot 2022-07-07 at 2 24 55 PM

What's next?
- Support AWS assume role.
- Update front end to show account id in the resource table.

### CODE REVIEW

File: pkg/provider/aws/autoscaling_test.go [hunk 1]
```diff
@@ -17,8 +17,9 @@ func TestFetchAutoscalingGroup(t *testing.T) {
 
 	testingutil.AssertResourceCount(t, resources, "", 2)
 	testingutil.AssertResourceFilteredCount(t, resources, 1, testingutil.ResourceFilter{
-		Type:   "autoscaling.AutoScalingGroup",
-		Region: defaultRegion,
+		AccountId: ctx.accountId,
```
[the remark this answers is not in the document]
Comment (RemyDeWolf) [reply pkg/provider/aws/autoscaling_test.go:20]: very good point, I'll do that

File: pkg/provider/aws/provider.go [hunk 2]
```diff
@@ -54,13 +54,15 @@ func (p *Provider) converterFor(resourceType string) resourceconverter.ResourceC
 	region := p.region.ID()
 	if mapping.UseMapConverter {
 		return &resourceconverter.MapConverter{
+			AccountId:    p.accountId,
```
[the remark this answers is not in the document]
Comment (RemyDeWolf) [reply pkg/provider/aws/provider.go:57]: Good point. I will update `Provider.Id` to be `Provider.AccountId`

File: cmd/root.go [hunk 3]
```diff
@@ -95,6 +99,7 @@ their cloud accounts.`,
 	flags.StringVarP(&rO.config, "config", "c", "", "Config file (default is https://github.com/run-x/cloudgrep/blob/main/pkg/config/config.yaml)")
 	flags.StringVar(&rO.bind, "bind", "", "Host to bind on")
 	flags.StringVarP(&rO.regions, "regions", "r", "", "Comma separated list of regions to scan, or \"all\"")
+	flags.StringVarP(&rO.profiles, "profiles", "", "", "Comma separated list of AWS profiles to scan.")
```
Comment (rabbitfang) [inline cmd/root.go:102]: suggestion(non-blocking): We can use `flags.StringSliceP` to split the values automatically, and to support multiple `--profiles` options (e.g. `--profiles=one,two --profiles=three`). Ditto for `--regions`, but that can be left alone as it's not part of this PR.

File: demo/setup/update_demo.go [hunk 4]
```diff
@@ -45,24 +46,40 @@ func main() {
 	}
 
 	//copy the tags from the ec2.instance to their ec2.NetworkInterface
-	for _, instance := range input.getResources(map[string]string{"core.type": "ec2.Instance"}) {
+	for _, instance := range input.getResources(map[string]interface{}{"core.type": "ec2.Instance"}) {
```
Comment (rabbitfang) [inline demo/setup/update_demo.go:49]:
```suggestion
	for _, instance := range input.getResources(map[string]any{"core.type": "ec2.Instance"}) {
```

suggestion(non-blocking): In 1.18, `any` is preferred over `interface{}`.

Ditto for the rest of this file.

File: pkg/config/config.go [hunk 5]
```diff
@@ -88,5 +92,53 @@ func LoadFromFile(file string) (Config, error) {
...
 	err = yaml.UnmarshalStrict(data, &cfg)
-	return cfg, err
+	if err != nil {
+		return Config{}, err
+	}
+	return cfg, nil
+}
+
+//Load will load the config files making sure that the options override are correct
+func (c Config) Load() (Config, error) {
```
Comment (rabbitfang) [inline pkg/config/config.go:102]: suggestion(non-blocking): add to the doc comment that the returned Config should be used in place of the original, since the original will go unmodified (due to the value receiver)

**Review by rabbitfang — APPROVED** [review]
LGTM. My only suggestions are non-blocking

File: cmd/root.go (hunk 3 above)
Comment (RemyDeWolf) [reply cmd/root.go:102]: this is good, I forgot about this

File: pkg/config/config.go (hunk 5 above)
Comment (RemyDeWolf) [reply pkg/config/config.go:102]: good find, I'll use a pointer to avoid this possible confusion
